packages/docs/versioned_docs/version-2.0.0-alpha-3/protocol/03-epoch-key.md

```diff
@@ -4,18 +4,19 @@ description: Definition of epoch key in UniRep
 
 # Epoch Key
 
-* _Epoch keys_ are temporary personas which users use them to interact with others.
-* Instead of giving attestations to an `identityCommitment`, a random-value-like `epochKey` is the **receiver** of an attestation.
-* Epoch key is computed by
+* Unique epoch keys are generated for each user in each epoch.
+* They are the users' temporary personas, allowing anonymous interaction with others.
+* Instead of giving attestations to an `identityCommitment`, an ever-changing, random-value-like `epochKey` is the **receiver** of an attestation.
+* An epoch key is computed by:
 
 ```typescript
 hash(identityNullifier, atteserId, epoch, nonce) % 2^epochTreeDepth
 ```
 
-where `nonce` and be any value between `0` and `numEpochKeyNoncePerEpoch - 1`, so that a user can have `numEpochKeyNoncePerEpoch` epoch keys per epoch.
+where `nonce` can be any value between `0` and `numEpochKeyNoncePerEpoch - 1`, so that a user can have `numEpochKeyNoncePerEpoch` epoch keys per epoch.
 
-* Only the user knows his `identityNullifier` so only he knows if he is receiving an attestation, others would see an attestation attesting to a random value.
-* In the [epoch key proof](../circuits-api/circuits#epoch-key-proof) circuit a user can prove that they know the `epochKey` and can receive and process attestations given to this `epochKey`.
+* Only a user knows their `identityNullifier`, so only they know if they have received an attestation; others see an attestation given to a random value.
```

Comment:
How does a user know their identityNullifier? I think this is the same question I have about a user's identityCommitment -- how is the user storing and using these values when they interact with the attester application?

Reply:
The user must save the `identityNullifier` and treat it like a private key. The `identityCommitment` is derived from the `identityNullifier`.

Users typically store these in browser `localStorage` and/or encrypted in a file on disk.